website/static/js/folderPickerNodeConfig.js

```diff
@@ -350,6 +350,8 @@ var FolderPickerViewModel = oop.defclass({
             self.nodeHasAuth(false);
             self.cancelSelection();
             self.currentDisplay(null);
+            $(self.folderPicker).html('');
```

Comment:
It might make sense to implement a `FolderPicker#destroy` method (which delegates to treebeard's destroy method rather than removing the html manually.